torchvision/models/detection/anchor_utils.py

```diff
@@ -100,7 +100,10 @@ def grid_anchors(self, grid_sizes, strides):
         anchors = []
         cell_anchors = self.cell_anchors
         assert cell_anchors is not None
-        assert len(grid_sizes) == len(strides) == len(cell_anchors)
+
+        if len(grid_sizes) != len(strides) != len(cell_anchors):
```

Comment:
I don't think this equivalent to the previous implementation.
Quick sanity check:
```python
In [14]: 1 == 1 == 2
Out[14]: False

In [15]: 1 != 1 != 2
Out[15]: False
```
So the error messages will not be raised on some cases.

Can you do instead something like
```python
if not (... and .. and ...):
```

Reply:
Oh yes ! What a rookie mistake 😢  Let me fix that up